client/my-sites/themes/events/theme-showcase-tracks.ts

```diff
@@ -51,7 +51,7 @@ const getThemeShowcaseEventRecorder = (
 				is_active_theme: isActiveTheme( themeId ),
 				style_variation: styleVariation,
 				results_rank: adjustedPosition,
-				results: themes.map( property( 'id' ) ).join(),
+				results: themes?.map( property( 'id' ) ).join() ?? [],
```

Comment:
Given this is typescript isn't `themes` always an array, and hence map doesn't need to be optionally chained?

Reply:
The problem with TypeScript is that it just works on build, during runtime you really can get an empty or null value. TypeScript would just avoid that when you use the type you always supply it.